erpnext/hr/report/monthly_attendance_sheet/monthly_attendance_sheet.py

```diff
@@ -120,137 +138,258 @@ def get_chart_data(emp_att_map, days):
 
 		if not filters.summarized_view:
 			row += emp_status_map
-
 		if filters.summarized_view:
 			row += [total_p, total_l, total_a, total_h, total_um]
 
+		conditions, filters = get_conditions(filters)
 		if not filters.get("employee"):
 			filters.update({"employee": emp})
 			conditions += " and employee = %(employee)s"
 		elif not filters.get("employee") == emp:
 			filters.update({"employee": emp})
 
 		if filters.summarized_view:
-			leave_details = frappe.db.sql("""select leave_type, status, count(*) as count from `tabAttendance`\
-				where leave_type is not NULL %s group by leave_type, status""" % conditions, filters, as_dict=1)
-
-			time_default_counts = frappe.db.sql("""select (select count(*) from `tabAttendance` where \
-				late_entry = 1 %s) as late_entry_count, (select count(*) from tabAttendance where \
-				early_exit = 1 %s) as early_exit_count""" % (conditions, conditions), filters)
-
-			leaves = {}
-			for d in leave_details:
-				if d.status == "Half Day":
-					d.count = d.count * 0.5
-				if d.leave_type in leaves:
-					leaves[d.leave_type] += d.count
-				else:
-					leaves[d.leave_type] = d.count
-
-			for d in leave_list:
-				if d in leaves:
-					row.append(leaves[d])
-				else:
-					row.append("0.0")
-
-			row.extend([time_default_counts[0][0],time_default_counts[0][1]])
+			get_leave_type_and_total_leave_taken(row, conditions, filters, emp, leave_list=leave_list)
+			get_late_entry_and_earl_exit_count(row, conditions, filters)
+
 		emp_att_map[emp] = emp_status_map
 		record.append(row)
-
 	return record, emp_att_map
 
+def get_late_entry_and_earl_exit_count(row, conditions, filters):
+	time_default_counts = frappe.db.sql("""select (select count(*) from `tabAttendance` where \
+		late_entry = 1 %s) as late_entry_count, (select count(*) from tabAttendance where \
+		early_exit = 1 %s) as early_exit_count""" % (conditions, conditions), filters)
+
+	row.extend([time_default_counts[0][0],time_default_counts[0][1]])
+
+def get_leave_type_and_total_leave_taken(row, conditions, filters, emp, leave_list=None):
+	leave_details = frappe.db.sql("""select leave_type, status, count(*) as count from `tabAttendance`\
+		where leave_type is not NULL %s group by leave_type, status""" % conditions, filters, as_dict=1)
+
+	leaves = {}
+	for d in leave_details:
+		if d.status == "Half Day":
+			d.count = d.count * 0.5
+		if d.leave_type in leaves:
+			leaves[d.leave_type] += d.count
+		else:
+			leaves[d.leave_type] = d.count
+
+
+	for d in leave_list:
+		d = d.replace(":Float:120", "")
+		print(d)
+		if d in leaves:
+			row.append(leaves[d])
+		else:
+			row.append("0.0")
+
+def get_totals(status, status_map, attendance_detail):
+
+	present = absent = leave = holiday = unmarked= 0.0
+	if status and attendance_detail:
+		if status in ["Weekly Off", "Holiday"]:
+			holiday += 1
+		if status_map[status]['is_present']:
+			present += 1
+		elif status_map[status]['is_leave']:
+			leave += 1
+		elif status_map[status]['is_half_day']:
```

Comment:
```suggestion
		if status_map[status]["is_present"]:
			present += 1
		elif status_map[status]["is_leave'"]:
			leave += 1
		elif status_map[status]["is_half_day"]:
```

Try using uniform quote styles throughout one file.